pkg/consensus/consensus.go

```diff
@@ -335,6 +360,32 @@ events:
 	close(s.finished)
 }
 
+func (s *service) resetWatchdogTimer(threshold time.Duration) {
+	lastTimestamp := int64(s.lastTimestamp)
+	now := time.Now().UnixMilli()
+	delta := time.Millisecond * time.Duration(lastTimestamp-now)
+	resetAfter := delta
+	for {
+		resetAfter += threshold
+		if resetAfter > 0 {
```

Comment:
Huh, this solves block 0 btw.

Reply:
Yes, it was done to solve this problem. And to be able to implement [this optimisation](https://github.com/nspcc-dev/neo-go/pull/2562#issuecomment-1168640993) also. I think it's OK to have this overhead for genesis block, it happens once per the whole node lifetime. Or do we need a separate logic for genesis?